src/journal/managed-journal-file.c

```diff
@@ -538,6 +539,17 @@ int managed_journal_file_open_reliably(
         /* The file is corrupted. Rotate it away and try it again (but only once) */
         log_warning_errno(r, "File %s corrupted or uncleanly shut down, renaming and replacing.", fname);
 
+        if (!template) {
+                /* The file is corrupted and no template is specified. Try opening it read-only as the
+                 * template before rotating to inherit its sequence number and ID. */
+                r = managed_journal_file_open(-1, fname, O_RDONLY, file_flags, 0, compress_threshold_bytes, NULL,
```

Comment:
i guess this doesn't matter much, but I think the open_flags should be specified as:

```c
…, (open_flags & ~(O_ACCMODE|O_CREAT|O_EXCL)) | O_RDONLY, …
```